tests/lib/collectors/grr_hunt.py

```diff
@@ -234,5 +235,56 @@ def testBadZipFileExtractHuntResults(self, mock_extract, mock_remove):
     mock_remove.assert_not_called()
 
 
+class GRRHuntOsqueryDownloader(unittest.TestCase):
+  """Tests for the GRR Osquery hunt downloader."""
+
+  @mock.patch('grr_api_client.api.InitHttp')
+  def setUp(self, mock_InitHttp):
+    self.mock_grr_api = mock.Mock()
+    mock_InitHttp.return_value = self.mock_grr_api
+    self.test_state = state.DFTimewolfState(config.Config)
+    self.grr_hunt_downloader = grr_hunt.GRRHuntOsqueryDownloader(
+        self.test_state)
+    self.grr_hunt_downloader.SetUp(
+        hunt_id='H:12345',
+        reason='random reason',
+        grr_server_url='http://fake/endpoint',
+        grr_username='admin',
+        grr_password='admin',
+        approvers='approver1,approver2',
+        verify=False
+    )
+    self.grr_hunt_downloader.output_path = '/tmp/test'
+
+  def testInitialization(self):
+    """Tests that the collector is correctly initialized."""
+    self.assertEqual(self.grr_hunt_downloader.hunt_id, 'H:12345')
+
+  @mock.patch('dftimewolf.lib.collectors.grr_hunt.GRRHuntOsqueryDownloader._GetAndWriteResults')  # pylint: disable=line-too-long
+  def testProcess(self, mock_get_write_results):
+    """Tests that hunt results are downloaded to the correct path."""
+    self.mock_grr_api.Hunt.return_value.Get.return_value = \
+        mock_grr_hosts.MOCK_HUNT
+    self.grr_hunt_downloader.Process()
+    mock_get_write_results.assert_called_with(mock_grr_hosts.MOCK_HUNT,
+                                              '/tmp/test')
+
+  @mock.patch('grr_api_client.hunt.Hunt.ListResults')
+  def testGetAndWriteResults(self, mock_list_results):
+    """Tests the GetAndWriteReslts function."""
+    mock_result = mock.MagicMock()
+    mock_result.payload = mock.MagicMock(spec=osquery_flows.OsqueryResult)
+    mock_list_results.return_value = [mock_result]
+    mock_client = mock.MagicMock()
+    mock_client.data.os_info.fqdn = 'TEST'
+    self.mock_grr_api.SearchClients.return_value = [mock_client]
+
+    results = self.grr_hunt_downloader._GetAndWriteResults(  # pylint: disable=protected-access
+        mock_grr_hosts.MOCK_HUNT, '/tmp/test')
+
+    self.assertEqual(len(results), 1)
+    self.assertEqual(results[0][0], 'test')
+    self.assertEqual(results[0][1], '/tmp/test/test.csv')
+
```

Comment:
I'm generally a fan of having tests for error scenarios, is it possible to adds some in? Hosts not found, flows not found for example.

Reply:
> I'd like to see some extra tests, but aside from that, LGTM. I'd like to hold off on merging until the PR actions get fixed though.

Thanks, will add some extra tests and hold off on the merge as suggested.

Reply:
Added an extra test for an incorrect flow type.